appdaemon/state.py

```diff
@@ -101,17 +101,16 @@ def terminate(self):
             # In the case of a quick_start parameter,
             # start the clock immediately if the device is already in the new state
             #
-            if "immediate" in kwargs and kwargs["immediate"] is True:
-                __duration = 0 # run it immediately
+            if kwargs.get("immediate"):
+                __duration = kwargs.get("duration", 0) # run it immediately
                 __new_state = None
                 __attribute = None
                 run = False
-                
+
                 if entity is not None and entity in self.state[namespace]:
                     run = True
-                    
-                    if "attribute" in kwargs:
-                        __attribute = kwargs["attribute"]
+                    __attribute = kwargs.get("attribute")
```

Comment:
Same thing as before, `attribute` is not always supplied, so should be `__attribute = kwargs.get("attribute", None)`

Reply:
`kwargs.get("attribute", None)` and `kwargs.get("attribute")` both return `None`, that is the default value of e.g., `dict.pop` and `dict.get`.

Reply:
In this case, None is the default return from `get()` if the key is not present. So:
```
kwargs.get("attribute")
```
and
```
kwargs.get('attribute', None)
```
are exactly the same.